test/semantics/select-rank.f90

```diff
@@ -0,0 +1,115 @@
+!Tests for SELECT RANK Construct(R1148)
+
+   !No error expected
+   subroutine CALL_ME(x)
+    implicit none
+    integer :: x(..)
+    SELECT RANK(x)
+    RANK (0)
+      print *, "PRINT RANK 0"
+    RANK (1)
+      print *, "PRINT RANK 1"
+    END SELECT
+   end
+
+   subroutine CALL_ME9(x)
+    implicit none
+    integer :: x(..)
+    boo: SELECT RANK(x)
+    RANK (1+0)
+      print *, "PRINT RANK 1"
+    END SELECT boo
+   end subroutine
+
+   !Error expected
+   subroutine CALL_ME2(x)
+    implicit none
+    integer :: x(..)
+    integer :: y(3)
+    !ERROR: Selector is not an assumed-rank array variable
+    SELECT RANK(y)
+    RANK (0)
+      print *, "PRINT RANK 0"
```

Comment:
It would be a better test here if `y` were to be used in some context that required it to be a scalar variable.  Similarly, in the other `RANK` cases, use the *associate-name* in some context that will elicit an error if the required rank were incorrect.

Reply:
I have added some tests are they as expected ones ?

Reply:
No.  Instead of just having a `PRINT` statement, try using the `SELECT RANK` construct entity in the test in some way that would work *only when the construct entity has the expected rank*, using something with a compile-time test of the rank.

```
SELECT RANK(x)
RANK(0)
  j = INT(0, KIND=MERGE(KIND(0), -1, RANK(x) == 0)) ! will fail when RANK(x) is not zero here
END SELECT
```

Reply:
For the patch submitted(currently passed a non zero value to pass the drone.io build) , test Fails with below error, when the `rank` is `0`, which is supposed to fail instead when `rank` is `not zero`.
`_../f18/test/semantics/select-rank.f90:39:11: error: 'kind=' argument must be a constant scalar integer whose value is a supported kind for the intrinsic result type
`

`
        j = INT(0, KIND=MERGE(KIND(0), -1, RANK(x) == 0)) ! will fail when RANK(x) is not zero 
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
_`
Is this an issue with the `RANK` construct or `select rank` ?

Reply:
That seems to indicate a problem with the semantics or name resolution of  x` in the `RANK`.  If the rank of `x` in that context were zero, as it should be, the `KIND=` argument would have had a valid value.  You can test that hypothesis by changing `==` to `/=` in that expression and seeing whether it (incorrectly) does not emit an error.

Reply:
Conformed for above discussed test with suggested changes,
`j = INT(0, KIND=MERGE(KIND(0), -1, RANK(x) /= 0))   ` `!Is evaluating to true, `which means `RANK` is not resolving properly the value of `x`.